src/net_processing.cpp

```diff
@@ -235,32 +235,48 @@ struct Peer {
+     *  transactions with a lower fee rate than this. See BIP133. */
     CAmount m_fee_filter_sent{0};
-    /** Timestamp to send the next BIP133 `feefilter` message sent to the peer. */
+    /** Timestamp after which we will send the next BIP133 `feefilter` message
+      * to the peer. */
     std::chrono::microseconds m_next_send_feefilter{0};
 
     struct TxRelay {
+        /** Protects m_bloom_filter and the m_relay_txs flag. */
         mutable RecursiveMutex m_bloom_filter_mutex;
-        // We use m_relay_txs for two purposes -
-        // a) it allows us to not relay tx invs before receiving the peer's version message
-        // b) the peer may tell us in its version message that we should not relay tx invs
-        //    unless it loads a bloom filter.
+        /** Whether the peer wishes to receive transaction announcements.
+         *
+         * This is initially set based on the fRelay flag in the received
+         * `version` message. If initially set to false, it can only be flipped
+         * to true if we have offered the peer NODE_BLOOM services and it sends
+         * us a `filterload` or `filterclear` message. See BIP37. */
         bool m_relay_txs GUARDED_BY(m_bloom_filter_mutex){false};
+        /** A bloom filter for which transactions to announce to the peer. See BIP37. */
         std::unique_ptr<CBloomFilter> m_bloom_filter PT_GUARDED_BY(m_bloom_filter_mutex) GUARDED_BY(m_bloom_filter_mutex){nullptr};
 
+        /** Protects m_tx_inventory_known and the m_send_mempool flag */
         mutable RecursiveMutex m_tx_inventory_mutex;
+        /** A filter of all the txids and wtxids that the peer has announced to
+         *  us or we have announced to the peer. We use this to avoid announcing
+         *  the same txid/wtxid to a peer that already has the transaction. */
         CRollingBloomFilter m_tx_inventory_known_filter GUARDED_BY(m_tx_inventory_mutex){50000, 0.000001};
-        // Set of transaction ids we still have to announce.
-        // They are sorted by the mempool before relay, so the order is not important.
+        /** Set of transaction ids we still have to announce (txid for
+         *  non-wtxid-relay peers, wtxid for wtxid-relay peers). We use the
+         *  mempool to sort transactions in dependency order before relay, so
+         *  this does not have to be sorted. */
         std::set<uint256> m_tx_inventory_to_send;
-        // Used for BIP35 mempool sending
+        /** Whether the peer has requested us to send our complete mempool. Only
+         *  permitted if the peer has NetPermissionFlags::Mempool. See BIP35. */
         bool m_send_mempool GUARDED_BY(m_tx_inventory_mutex){false};
-        // Last time a "MEMPOOL" request was serviced.
+        /** The last time a BIP35 `mempool` request was serviced. */
         std::atomic<std::chrono::seconds> m_last_mempool_req{0s};
+        /** The next time after which we will send an `inv` message containing
```

Comment:
Nit: it also determines sending mempool response through `fSendTrickle`
```

// Respond to BIP35 mempool requests
if (fSendTrickle && peer->m_tx_relay->m_send_mempool) {
```

Reply:
Arguably, this field could be moved out of `TxRelay` then. Or perhaps we should use another timer for mempool.

Reply:
> Nit: it also determines (through fSendTrickle) sending mempool response

The response to a mempool message is one or more inv announcement messages for transactions in the mempool. I think the comment is correct.

> Arguably, this field could be moved out of TxRelay then. Or perhaps we should use another timer for mempool.

I think it'd be better to simply remove support for the `mempool` message. I think it's unlikely that many people are using it, and it'd be better to provide this functionality (exposing the contents of the mempool) through rpc rather than a 'private' p2p extension.